pool_dynamics_test.go

```diff
@@ -265,38 +267,32 @@ func TestPoolAffinity(t *testing.T) {
 				Size:    float64(baseStatSize) * float64(10),
 			}
 
-			// variedStats := util.NodeStats{
-			// 	Start:   time.Now().Add(-time.Second * 2),
-			// 	Latency: float64(baseStatLatency) / (float64(10) + (1 + statVarianceFactor)),
-			// 	Size:    float64(baseStatSize) * float64(10) * (1 + statVarianceFactor),
-			// }
-
 			ch.RecordSuccesses(t, goodNodes, baseStats, 100)
 			ch.RecordSuccesses(t, badNodes, baseStats, 10)
 
 			ch.CaboosePool.DoRefresh()
 		}
 
-		// for _, i := range ch.CabooseAllNodes.Nodes {
-		// 	fmt.Println(i.URL, i.Priority(), i.PredictedLatency)
-		// }
+		rerouteCount := 0
 
-		// Get the candidate nodes for a few cids from our formed cid list using
-		// the affinity of each cid.
-		for i := 0; i < 10; i++ {
-			rand.New(rand.NewSource(time.Now().Unix()))
-			idx := rand.Intn(len(cidList))
-			c := cidList[idx]
+		// Get the candidate nodes for each cid in the cid list to see if it's been rerouted to a new node.
+		for _, c := range cidList {
 			aff := ch.Caboose.GetAffinity(ctx)
 			if aff == "" {
 				aff = fmt.Sprintf(blockPathPattern, c)
 			}
 			nodes, _ := ch.CabooseActiveNodes.GetNodes(aff, ch.Config.MaxRetrievalAttempts)
 
-			// We expect that the candidate nodes are part of the "good nodes" list.
-			assert.Contains(t, goodNodes, nodes[0])
+			for _, n := range badNodes {
+				n := n
+				if n.URL == nodes[0].URL {
+					rerouteCount++
+				}
```

Comment:
should 'reroute' be incremented when this doesn't match?

Reply:
I believe reroute is meant to keep track of how often caboose returns a new joining node as the candidate node to fetch a cid in which case caboose is "rerouting" where it fetches that cid from. In that case, the increment here is good imo.

Reply:
Yeah what Amean said.

Reply:
We're iterating over bad nodes here.